qutebrowser/browser/commands.py

```diff
@@ -822,22 +876,36 @@ def zoom(self, zoom: int = None, count=None):
         message.info("Zoom level: {}%".format(level), replace=True)
 
     @cmdutils.register(instance='command-dispatcher', scope='window')
-    def tab_only(self, prev=False, next_=False):
+    def tab_only(self, prev=False, next_=False, force=False):
         """Close all tabs except for the current one.
 
         Args:
             prev: Keep tabs before the current.
             next_: Keep tabs after the current.
+            force: Avoid confirmation for pinned tabs.
         """
         cmdutils.check_exclusive((prev, next_), 'pn')
         cur_idx = self._tabbed_browser.currentIndex()
         assert cur_idx != -1
 
+        def _to_close(i):
```

Comment:
(if we consolate the loops below, this does not apply but):
The reason I split this out is to avoid copying the variable set logic for both of the below loops. I can copy the set into both of them, but I feel like that would be more messy, thoughts?